src/compiler/types.ts

```diff
@@ -1525,7 +1526,9 @@ module ts {
     export interface SymbolLinks {
         target?: Symbol;                    // Resolved (non-alias) target of an alias
         type?: Type;                        // Type of value symbol
-        declaredType?: Type;                // Type of class, interface, enum, or type parameter
+        declaredType?: Type;                // Type of class, interface, enum, type alias, or type parameter
+        typeParameters?: TypeParameter[];   // Type parameters of type alias (undefined if non-generic)

```

Comment:
i would call this typeAliasTypeParameter.  That way it's clear from use that it's only intended for Type aliases.  "typeParameters" on its own is very vague, and makes it unclear at the use site that it's only for these cases, and not for anything else with type parameters.  (same with instantiations).